theseus/core/vectorizer.py

```diff
@@ -33,6 +34,12 @@ def _varinfo(var) -> str:
     )
 
 
+class _VectorizationMode(Enum):
+    ERROR = 0
+    WEIGHTED_ERROR = 1
+    FULL = 2
```

Comment:
Might be helpful to leave some brief comments about each mode and what it does.

Reply:
I'll add some comments.